src/chatClient.cpp

```diff
@@ -594,23 +594,25 @@ void Client::onRequestFinish(::mega::MegaApi* apiObj, ::mega::MegaRequest *reque
     case ::mega::MegaRequest::TYPE_FETCH_NODES:
     {
         api.sdk.pauseActionPackets();
-        auto state = mInitState;
-        char* pscsn = api.sdk.getSequenceNumber();
-        std::string scsn;
-        if (pscsn)
-        {
-            scsn = pscsn;
-            delete[] pscsn;
-        }
-        std::shared_ptr<::mega::MegaUserList> contactList(api.sdk.getContacts());
-        std::shared_ptr<::mega::MegaTextChatList> chatList(api.sdk.getChatList());
 
         auto wptr = weakHandle();
-        marshallCall([wptr, this, state, scsn, contactList, chatList]()
+        marshallCall([wptr, this]()
         {
             if (wptr.deleted())
                 return;
 
+            char* pscsn = api.sdk.getSequenceNumber();
```

Comment:
What's the reason to capture all this data inside the marshal call? despite the action packets are paused, the app sharing the MegaApi instance may do requests in the meantime, which may result on different set of contacts/chats.